docs/DeveloperGuide.md

```diff
@@ -155,6 +158,36 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Assignment Feature
+
+#### Current Implementation
+The `Assignment` class encapsulates the current Assignment feature and composes of `Description`, `Status` and `DueDate` class.
+
+It implements the operation `Assignment#isSameAssignment(Assignment assignment)` to check for duplicate assignments. Currently, assignments are similar if they have the same description and this check is case-insensitive. This is because each student is under one module and having a similarly named assignment within the same module is less likely.
+
+Next, the current available `Status` of `Assignment` are "PENDING" and "COMPLETED". Since the type of `Status` are fixed, the `Status` class contains a `enumeration StatusType` to store the valid values. The use of static methods `Status#createCompletedStatus()` and `Status#createPendingStatus()` initialises the "COMPLETED" and "PENDING" `Status` respectively. Meanwhile, the constructor is set to private to prevent instantiation through inheritance. 
```

Comment:
`COMPLETED` instead of "COMPLETED"

Reply:
maybe "the constructor is set to private to prevent..." could specify constructor of status